lib/IRGen/GenFunc.cpp

```diff
@@ -1203,19 +1318,29 @@ static llvm::Value *emitPartialApplicationForwarder(IRGenModule &IGM,
 
   IRGenFunction subIGF(IGM, fwd);
   if (origType->isAsync()) {
-    subIGF.setupAsync(
-        Signature::forAsyncEntry(IGM, outType).getAsyncContextIndex());
+    auto asyncContextIdx =
+        Signature::forAsyncEntry(IGM, outType).getAsyncContextIndex();
+    asyncLayout.emplace(irgen::getAsyncContextLayout(
+        IGM, origType, substType, subs, /*suppress generics*/ false,
+        FunctionPointer::Kind(
+            FunctionPointer::BasicKind::AsyncFunctionPointer)));
+
+    subIGF.setupAsync(asyncContextIdx);
 
-    auto *calleeAFP = staticFnPtr->getDirectPointer();
+    //auto *calleeAFP = staticFnPtr->getDirectPointer();
     LinkEntity entity = LinkEntity::forPartialApplyForwarder(fwd);
-    auto size = Size(0);
     assert(!asyncFunctionPtr &&
            "already had an async function pointer to the forwarder?!");
-    asyncFunctionPtr = emitAsyncFunctionPointer(IGM, fwd, entity, size);
+    emitAsyncFunctionEntry(subIGF, *asyncLayout, entity, asyncContextIdx);
+    asyncFunctionPtr =
+        emitAsyncFunctionPointer(IGM, fwd, entity, asyncLayout->getSize());
+    // TODO: if calleeAFP is definition:
+#if 0
```

Comment:
Should this still be here?

Reply:
Yes. We will want to reinstate the old code conditionally later. This serves as a reminder for me how-to do it.